server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -318,8 +339,7 @@ Settings builtAnalyzerSettings() {
         return analyzerSettings;
     }
 
-    private static void applyAndValidateStorageSettings(Map<String, Object> mapping,
-                                                        AnalyzedColumnDefinition<Object> definition) {
+    public static void docValuesSpecifiedAndDisabled(AnalyzedColumnDefinition<Object> definition) {
```

Comment:
we used to call `applyAndValidateStorageSettings` only in `toMapping` -  but we don't need to compute the whole mapping on the new code branch - thus moved this to a separate method so that it can be part s of validation - this code not only compute hasDocValue flag but also validates columnStorage properties so moved it t validate. See https://github.com/crate/crate/pull/13238/commits/1ad1189bf26f1f42df709f257714b2b6f6738351 for details